### DISCUSSION TITLE
Native validation for a single checkbox

### DISCUSSION THREAD

**Discussion by phuoccss** [adobe/react-spectrum#6137] [General]
Is there a way to use native validation with a single react-aria-components's Checkbox (standalone, not in a CheckboxGroup), like shown here https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_checkbox_required?

**Comment by snowystinger** [comment]
Is it not working standalone? Can you provide a codesandbox or stackblitz reproducing the issue you're running into?

**Reply by phuoccss** [reply]
You can take a look here https://codesandbox.io/p/devbox/old-butterfly-lm8wm4?file=%2Fsrc%2FApp.tsx%3A45%2C12. The one on top is native checkbox.

From the [docs](https://react-spectrum.adobe.com/react-aria/Checkbox.html#validation), Checkbox only accepts `isInvalid` props.  It does not accept `<FieldError />` as children like other form fields. My guess is this is due to the DOM structure.

A workaround is to use CheckboxGroupField with only 1 Checkbox, but not sure if it's recommended accessibility-wise.

**Reply by snowystinger** [reply]
Thanks so much for sharing that. I'll bring it up with the team.

As for now, I think using the group with only 1 checkbox is fine.

**Reply by LFDanLu** [reply]
For some more background, we specifically suppress the default browser error UI here: https://github.com/adobe/react-spectrum/blob/12f2f4e57689025d02faef76682c923262ce88f6/packages/%40react-aria/form/src/useFormValidation.ts#L70-L71 in favor of FieldError (RAC) and our own Spectrum styled validation styling: https://react-spectrum.adobe.com/react-spectrum/forms.html#built-in-validation. Perhaps there could be an option to disable that `preventDefault` call, especially if you were just using the hooks to build your checkbox, but we'd have to be careful that that tooltip doesn't appear with the FieldError since that would be quite odd. Not sure if there are any other particular reasons against allowing the default browser errors from appearing other than that though

**Reply by phuoccss** [reply]
I was not clear enough. What I wanted to achieve was using `FieldError` to display the native validation message. Problem is `Checkbox` does not accept `FieldError` as children, hence there's no way to display the error. Disabling `preventDefault` will make the browser tooltip show up but it is inconsistent with other form fields like `TextField`, `CheckboxGroup` etc.

**Reply by LFDanLu** [reply]
Ah I see, thank you for the clarification. If I remember correctly the team had considered supporting `FieldError` with individual Checkboxes in the past but were worried about if users would add a `FieldError` to each individual Checkbox within a CheckboxGroup. Perhaps we could only render the `FieldError` if the Checkbox isn't with a CheckboxGroup...

**Reply by phuoccss** [reply]
For `Checkbox`, props like `defaultSelected` or `isSelected` are ignored when used within a `CheckboxGroup`. So I guess it would be okay to do something similar for `FieldError`.

**Reply by LFDanLu** [reply]
Talked with the team, we'll want to add support for `FieldError`/`HelpText` (aka descriptions) to Checkbox, Radio, and Switch (Switch will only have support for `HelpText`). Will file an issue: https://github.com/adobe/react-spectrum/issues/6192

**Reply by phuoccss** [reply]
Thanks for looking into this.
